src/cil.ml

```diff
@@ -2079,25 +2077,20 @@ let truncateCilint (k: ikind) (i: cilint) : cilint * truncation =
     else
       truncate_unsigned_cilint i nrBits
 
+let mkCilintIk (ik:ikind) (i:cilint) : cilint =
+  fst (truncateCilint ik i)
+
 let mkCilint (ik:ikind) (i:int64) : cilint =
-  fst (truncateCilint ik (cilint_of_int64 i))
+ mkCilintIk ik (cilint_of_int64 i)
+
 
 (* Construct an integer constant with possible truncation *)
 let kintegerCilint (k: ikind) (i: cilint) : exp =
   let i', truncated = truncateCilint k i in
   if truncated = BitTruncation && !warnTruncate then
     ignore (warnOpt "Truncating integer %s to %s"
               (string_of_cilint i) (string_of_cilint i'));
-  let str =
-    let int64_min = cilint_of_int64 Int64.min_int in
-    let int64_max = cilint_of_int64 Int64.max_int in
-    (* if the resulting value can not be represented by an Int64, store its string representation *)
-    if Cilint.compare_cilint i' int64_min < 0 || Cilint.compare_cilint int64_max i' < 0 then (
-      Some (string_of_cilint i')
-    )
-    else None
-  in
-  Const (CInt64(int64_of_cilint i', k, str))
+  Const (CInt(i', k, None))
```

Comment:
Interestingly, it looks like yes. The CIL manual has this comment about keeping string representations to write `0xF` out as `0xF` instead of `15`, but it does already not do that (anymore) in our current version, and it does not look like that was changed recently.

Reply:
I guess we should just remove this second component then.

Reply:
I actually decided to go the other way and now include the string whenever possible. It will not be helpful for Goblint, but printing constants as they were written is still a nice feature.

Reply:
I guess that changes the meaning of the string back to what the old manual says. It also means that just parsing the string isn't so easy because it preserves formatting (?) although that won't really matter since Goblint should use the `cilint` anyway in all cases. Just that we should be careful not to ever try to parse the string anymore.

Are there cases when this string isn't available then? Otherwise it'd be nice to make it non-optional.

Reply:
> It also means that just parsing the string isn't so easy because it preserves formatting (?) although that won't really matter since Goblint should use the `cilint` anyway in all cases.

Exactly, the parsing of these strings in Goblint was just a crutch we needed because of the limitation to `int64`.

> Are there cases when this string isn't available then?

Yes, e.g., when constant folding was applied or a constant was manually constructed somewhere.